codegen/src/types/type_path.rs

```diff
@@ -76,6 +76,27 @@ impl TypePath {
             Self::Substitute(sub) => sub.parent_type_params(acc),
         }
     }
+
+    /// Gets the vector type parameter if the data is represented as `TypeDef::Sequence`.
+    ///
+    /// **Note:** Utilized for transforming `std::vec::Vec<T>` into slices `&[T]` for the storage API.
+    pub fn vec_type_param(&self) -> Option<TokenStream> {
+        match self {
+            TypePath::Type(ty) => {
+                match ty.ty.type_def() {
+                    TypeDef::Sequence(_) => {
```

Comment:
I think things like HashMap and HashSet and such also encode to a `TypeDef::Sequence`, so you'll probably also need to check that the `.path() == &["Vec"]` :)